### ISSUE TITLE
Swagger generation should be automatically part of the build if swagger annotations are included?

### ISSUE THREAD

**Issue by pzfreo** [ballerina-platform/ballerina-lang#7355]
**Description:**
If we have kubernetes or docker annotations in .bal files, then the k8s or dockerfile is automatically generated as part of the compilation. However, the same behaviour doesn't seem to happen with Swagger, and it is necessary to run "ballerina swagger export". It would be more consistent if the annotations caused the generation to happen

**Steps to reproduce:**
Add a swagger annotation to .bal and build.

**Affected Versions:**
All

**OS, DB, other environment details and versions:**
Any

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by praminda** [comment]
@pzfreo with swagger to ballerina, we generate services which has swagger annotations. So generating swagger file if a @ swagger annotation is present might confuse the users with automatically generated swagger file.

**Comment by praminda** [comment]
@pzfreo In docker and k8s scenario, generating dokerfile and other artifacts is the expected outcome of having the annotation. However reasons for having swagger annotation is not the same. 

We add swagger annotations to declare additional properties in Swagger/OAS spec which are not directly declarable in ballerina source. These information is to be used by ballerina to swagger utility.
**Ex**: `operation description, tags.`

If theres swagger annotations in a ballerina service, most probably this means this service was generated by our code generators using a swagger file. So there is a higher chance for a user to not expect a swagger file as an output of building auto generated ballerina service.

Introducing a specific annotation to enable this seems fine but generating the swagger file for any swagger annotation doesn't look right to me. Thoughts?

**Comment by NipunaMarcus** [comment]
@pzfreo 
I also agree with what @praminda said. We use swagger annotation to keep additional information related to OAS. 

We currently have a service client generation which has an annotation to enable generating a client to a service (https://ballerina.io/learn/by-example/client-generation.html). We can have a specific annotation such as that to enable generation of swagger file but is it necessary to have such compiler plugin as we have a compiler extensions ?

**Comment by rasika** [comment]
There are two types of developers;

(1) Contract first developers - e.g. swagger-to-ballerina
(2) Code first developers - e.g. ballerina-to-swagger

> If theres swagger annotations in a ballerina service, most probably this means this service was generated by our code generators using a swagger file. So there is a higher chance for a user to not expect a swagger file as an output of building auto generated ballerina service.

With the current status; we assume that the majority is the contract-first developers. Thus, we don't generate swagger file even swagger annotations are present. On the other-hand, Code-first developers has to explicitly execute `ballerina swagger export input.bal` in-order to generate the swagger file.

Focusing on either type of developers means we are under-serving the other type. Hence, I would like to suggest that we always generate the swagger file whenever the swagger annotations are present. Then we don't need to worry about contract-first or code-first since both developers are happy EOD. Perhaps for the contract-first approach, built swagger file might not be 100% equal to the initial swagger file, But that's fine and 100% equivalancy should be our ultimate goal.

**Comment by NipunaMarcus** [comment]
@rasika I understand your point. 
If we look in to the GRPC scenario, In GRPC scenario we have given a similar experience by generating a `.proto` file on compilation. `.proto` file is a must as it is used to generate the client code stub.  But in our case generating a swagger file is bit unnecessary because to generate client for a ballerina service we have a separate annotation. So generating a swagger file will only be for mostly for sharing purposes and this is not everyday scenario. So IMO for this developer can use the ballerina-to-swagger tool.
